mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMailboxInconsistenciesService.java

```diff
@@ -87,6 +104,29 @@ private OrphanMailboxDAOEntry(Mailbox mailbox) {
                     }
                 });
         }
+
+        /**
+         * We validate:
+         *  - That the mailbox record stil exists and that its path did not change
+         *  - That no corresponding path record exist
+         */
+        @Override
+        public Mono<Boolean> isStillPertinent(Context context, CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO pathV2DAO) {
+            return mailboxDAO.retrieveMailbox((CassandraId) mailbox.getMailboxId())
+                .filter(currentMailbox -> currentMailbox.generateAssociatedPath().equals(mailbox.generateAssociatedPath()))
+                .flatMap(currentMailbox -> pathV2DAO.retrieveId(currentMailbox.generateAssociatedPath())
+                    .map(entry -> !INCONSISTENCY_STILL_PRESENT)
+                    .switchIfEmpty(Mono.just(INCONSISTENCY_STILL_PRESENT)))
+                .filter(stillPresent -> stillPresent)
+                .switchIfEmpty(Mono.fromCallable(() -> {
+                    context.errors.incrementAndGet();
```

Comment:
Demeter law

Reply:
I'm not very at ease to have a Query method performing mutating operations.

Reply:
I felt I had to choose between plague and cholera here...

This report should be more natural in `fixInconsistencyIfNeeded` below. But then the responsibility of updated context is shared in 2 places.

I bet we'd be better off mutating the inconsistency and adding a `ConcurrentlyModifiedXXXInconsistency` doing this report upon `fix call`.

Reply:
> I bet we'd be better off mutating the inconsistency and adding a `ConcurrentlyModifiedXXXInconsistency` doing this report upon `fix call`.

I do not understand